src/notebooks/Python Examples/GSC23 MGnify Workshop.ipynb

```diff
@@ -0,0 +1,1529 @@
+    "\n",
+    "This is an interactive code notebook (a Jupyter Notebook).\n",
+    "To run this code, click into each cell and press the ▶ button in the top toolbar, or press `shift+enter`.\n",
+    "\n",
+    "### This notebook is divided into 3 sections: \n",
+    "\n",
+    "- Section 1 focuses on MGnify Analyses (metagenomic datasets and assemblies analysed by MGnify)\n",
+    "- Section 2 focuses on the MGnify Genomes resource (the genome catalogues)\n",
+    "- Bonus section allows you to practice writing your own code to query the API\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d35fcccb-bf90-4cdd-8c36-415e9b843469",
+   "metadata": {},
+   "source": [
+    "### Import packages\n",
+    "\n",
+    "Execute the code below to import the Python libraries required to run the examples presented in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ba4e90f6-7778-4d03-ae4f-05711d5d5c63",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Connection to MGnify API\n",
+    "from jsonapi_client import Session as APISession\n",
+    "from jsonapi_client import Modifier  # allows us to query specific values in given fields (e.g.: 'taxon-lineage').\n",
+    "import requests\n",
+    "from itertools import islice\n",
+    "\n",
+    "# Dataframes and display\n",
+    "import pandas as pd\n",
+    "\n",
+    "pd.set_option('display.max_columns', None)\n",
+    "pd.set_option('display.max_colwidth', None)\n",
+    "\n",
+    "# Plots\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "import plotly.graph_objects as go\n",
+    "%matplotlib inline \n",
+    "\n",
+    "# Data download\n",
+    "from urllib.request import urlretrieve\n",
+    "\n",
+    "# Warning verbosity\n",
+    "import warnings \n",
+    "warnings.filterwarnings(action=\"ignore\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "16229ed2-70b0-4e5a-a2ec-3f8d63a227e9",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Section 1: explore MGnify analysis results programmatically"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0310546e-6b7a-41c9-a42a-7a03f0a16ab0",
+   "metadata": {},
+   "source": [
+    "In the first portion of today's workhop you learned about MGnify's analysis of metagenomic data. In this exercise we will fetch and explore MGnify analysis results programmatically. These are just a few examples of the types of commands and operations that can be peformed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ce46df50-8a47-40d0-a0aa-499fa333b4ac",
+   "metadata": {},
+   "source": [
+    "#### Task 1: Browse the recently analysed studies and save them to a file\n",
+    "\n",
+    "Let's start by loading 50 most recently analysed studies and saving them to a file called ```studies.csv``` (it would take too long to load all studies for this example so we limit the number at 50):"
```

Comment:
```suggestion
    "Let's start by loading the 50 most recently analysed studies and saving them to a file called ```studies.csv``` (it would take too long to load all studies for this example so we limit the number at 50):"
```